packages/mco/hooks/application-set.ts

```diff
@@ -0,0 +1,133 @@
+import * as React from 'react';
+import { useDeepCompareMemoize } from '@odf/shared/hooks/deep-compare-memoize';
+import {
+  getAPIVersion,
+  getName,
+  getNamespace,
+  getUID,
+} from '@odf/shared/selectors';
+import { referenceForModel } from '@odf/shared/utils';
+import { useK8sWatchResources } from '@openshift-console/dynamic-plugin-sdk';
+import {
+  ACMPlacementModel,
+  ACMPlacementDecisionModel,
+  ArgoApplicationSetModel,
+} from '../models';
+import {
+  ArgoApplicationSetKind,
+  ACMPlacementDecisionKind,
+  ACMPlacementKind,
+  CustomAcmApplicationPlacementKind,
+} from '../types';
+import {
+  findDeploymentClusterNameUsingPlacement,
+  findApplicationSetFromPlacement,
+  createRefFromApplicationSet,
+  isApplicationPresent,
+} from '../utils';
+
+const resources = (name?: string, namespace?: string) => ({
+  applications: {
+    name: name,
+    namespace: namespace,
+    kind: referenceForModel(ArgoApplicationSetModel),
+    isList: true,
+    namespaced: true,
+  },
+  placements: {
+    namespace: namespace,
+    kind: referenceForModel(ACMPlacementModel),
+    isList: true,
+    namespaced: true,
+  },
+  placementDecisions: {
+    namespace: namespace,
+    kind: referenceForModel(ACMPlacementDecisionModel),
+    isList: true,
+    namespaced: true,
+  },
+});
+
+export const useWatchApplicationSetResources: UseK8sWatchResource = (
```

Comment:
nits:
just to shorten the name
```suggestion
export const useWatchAppSetResources: UseK8sWatchResource = (
```